SoC-2020-Ideas.md

```diff
@@ -143,20 +177,51 @@ One can start with using this labeling for making selected single
 command faster, for example `--contains` queries (as it was done for
 generation numbers).
 
+Before starting on this task, at the beginning of the GSoC, it might
+be good idea to check that interval labels would provide significant
+performance improvements at least in some cases (and if it is not the
+case, switch to working on generation numbers v2).  This _could_ be
+done with the help of "Reachability labels for version control
+graphs.ipynb" Jupyter Notebook on Google Colaboratory[[6][]].  It was
+created to answer such questions, though the exploration didn't get
+finished.  It would be possible with this notebook to at least find
+the amount of false negatives for min-post interval labeling for
+git.git or Linux kernel repo.
+
 Next task would be, time permitting, to make it possible to update the
 labeling without recomputing it from scratch, and to make it
-compatible with incremental update of the commit-graph file[7].
+compatible with incremental update of the commit-graph file[[7][],[3][]].
 
 References:
 
 1. <https://githubengineering.com/counting-objects/>
 2. <https://devblogs.microsoft.com/devops/supercharging-the-git-commit-graph-iii-generations/>
-3. <https://drive.google.com/open?id=1psMBVfcRHcZeJ7AewGpdoymrEfFVdXoK>
+3. <https://drive.google.com/open?id=1psMBVfcRHcZeJ7AewGpdoymrEfFVdXoK>  
+   <https://www.slideshare.net/JakubNarbski/graph-operations-in-git-version-control-system-how-the-performance-was-improved-for-large-repositories-how-can-it-be-further-improved>  
+   <https://speakerdeck.com/jnareb/graph-operations-in-git-slides-2019>
```

Comment:
Just a minor suggestion here. It's not so clear in the rendered version that there are three links here. So, it would be nice if we use a list here. Something like:

```suggestion
3.  - <https://drive.google.com/open?id=1psMBVfcRHcZeJ7AewGpdoymrEfFVdXoK>  
    - <https://www.slideshare.net/JakubNarbski/graph-operations-in-git-version-control-system-how-the-performance-was-improved-for-large-repositories-how-can-it-be-further-improved>  
    - <https://speakerdeck.com/jnareb/graph-operations-in-git-slides-2019>
```

Reply:
@sivaraam : Thanks, did something like that.